storage/remote/max_timestamp.go

```diff
@@ -34,8 +34,14 @@ func (m *maxGauge) Set(value float64) {
 	}
 }
 
-func (m *maxGauge) Get() float64 {
+func (m *maxTimestamp) Get() float64 {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
 	return m.value
 }
+
+func (m *maxTimestamp) Collect(c chan<- prometheus.Metric) {
+	if m.Get() > 0 {
```

Comment:
I'm afraid this makes cases like the one described in https://github.com/prometheus/prometheus/issues/14350 (prometheus was never able to reach the remote write due to a network/config issues/mistake) harder to catch, as https://github.com/prometheus/prometheus/blob/3d54bcc018df3e081758db3e13a49f9a2566c49b/documentation/prometheus-mixin/alerts.libsonnet#L210-L219
will no longer fire because `prometheus_remote_storage_queue_highest_sent_timestamp_seconds` is no longer initialized.
We could adapt the alert expression, but I think the cleanest solution would be to drop this condition.

cc @simonpasquier